index.html

```diff
@@ -92,10 +92,47 @@ <h2><dfn>SmartCardResourceManager</dfn> interface</h2>
           Promise&lt;SmartCardContext&gt; establishContext();
         };
       </pre>
+      <p>Methods on this interface complete asynchronously, queuing
+      work on the <dfn>smart card task source</dfn>.</p>
       <section>
         <h3><dfn>establishContext()</dfn> method</h3>
         <p>Requests a PC/SC context from the platform's PC/SC stack.</p>
-        <div class="issue">Write an algorithm for this method.</div>
+        The {{SmartCardResourceManager/establishContext()}} method steps are:
+        <ol>
+          <li>
+            If [=this=]'s [=relevant global object=]'s [=associated Document=] is
+            not [=allowed to use=] the [=policy-controlled feature=] named
+            "[=policy-controlled feature/smart-card=]", [=exception/throw=] a
+            "{{SecurityError}}" {{DOMException}}.
+          </li>
+          <li>Let |promise:Promise| be [=a new promise=].</li>
+          <li>Run the following steps [=in parallel=]:
+          <ol>
+            <li>Let |resourceManager:RESOURCEMANAGER| be a new instance of the
+              platform's [[PCSC5]] `RESOURCEMANAGER` class.</li>
+            <li>Invoke the `EstablishContext` method of |resourceManager| with
+              a "system" `Scope` parameter.</li>
+            <li>If the returned `RESPONSECODE` is not `SCARD_S_SUCCESS`, perform
+              the following steps:
+              <ol>
+                <li>Destroy |resourceManager|.</li>
+                <li>[=Queue a global task=] on the [=relevant global object=] of
+              [=this=] using the [=smart card task source=] to
+              [=reject=] |promise| with a corresponding {{SmartCardError}}.</li>
+              </ol>
+            </li>
+            <li>Otherwise, perform the following steps:
+              <ol>
+                <li>Let |context:SmartCardContext| be a {{SmartCardContext}}
+                  containing |resourceManager|.</li>
```

Comment:
Follow-up: Define how `context` manages the `resourceManager` it "contains".

Reply:
Ack.